### DISCUSSION TITLE
algorithm structure

### DISCUSSION THREAD

**Discussion by D19CrMei** [talmolab/sleap#810] [General]
What is the detailed algorithm structure of SLEAP network? For example, what does its activation function, backbone network, and convolutional layers look like? Where can I find out?

**Comment by roomrys** [comment]
Hi @D19CrMei,

SLEAP allows the option to use a few different networks such as our own legacy [leap](https://sleap.ai/develop/api/sleap.nn.architectures.leap.html) network, [unet](https://sleap.ai/develop/api/sleap.nn.architectures.unet.html) (the default), [hourglass](https://sleap.ai/develop/api/sleap.nn.architectures.hourglass.html), [resnet](https://sleap.ai/develop/api/sleap.nn.architectures.resnet.html), and pretrained_encoder (which uses a unet with a pretrained [encoder](https://sleap.ai/develop/api/sleap.nn.architectures.encoder_decoder.html)). This documentation is currently listed in our [Developer API](https://sleap.ai/develop/api.html) page on the SLEAP website. 

![image](https://user-images.githubusercontent.com/38435167/164751820-226aaf75-00a5-4170-8d2c-3ef8220302f3.png)
**Im: All the types of backbone networks SLEAP supports. Taken from the Model Configuration in the SLEAP GUI.**

We could and should improve upon the descriptions of the networks with a few diagrams for assistance - so thank you for bringing this to our attention. For further reading on the technical details of SLEAP, we released a paper ([SLEAP: Multi-animal pose tracking](https://www.nature.com/articles/s41592-022-01426-1)) last month which has a **Network architectures** subsection located near the end of the paper in **Methods**.

Let me know if this is what you were looking for.

Thanks,
Liezl

**Comment by D19CrMei** [comment]
I browsed the latest papers, but still couldn't find the specific composition of each module in sleep, such as the number of neural network layers and training methods.


Also, I want to train my own dataset, any suggestions for building the dataset?


| |
梅Dr.CrT
|
|
***@***.***
|
签名由网易邮箱大师定制
On 4/23/2022 00:22，Liezl ***@***.***> wrote：

Hi @D19CrMei,

SLEAP allows the option to use a few different networks such as our own legacy leap network, unet (the default), hourglass, resnet, and pretrained_encoder (which uses a unet with a pretrained encoder). This documentation is listed in our currently listed in our Developer API page on the SLEAP website.


Im: All the types of backbone networks SLEAP supports. Taken from the Model Configuration in the SLEAP GUI.

We could and should improve upon the descriptions of the networks with a few diagrams for assistance - so thank you for bringing this to our attention. For further reading on the technical details of SLEAP, we released a paper (SLEAP:Multi-animal pose tracking) last month which has a Network architectures subsection located near the end of the paper in Methods.

Let me know if this is what you were looking for.

Thanks,
Liezl

**Comment by roomrys** [comment]
@D19CrMei we have a good [tutorial](https://sleap.ai/tutorials/tutorial.html) on the website which should help you get going.
